lib/mongodb.js

```diff
@@ -734,8 +734,13 @@ MongoDB.prototype.buildSort = function(model, order) {
 };
 
 function buildNearFilter(query, near) {
+  var node = query.where;
+  near.key.forEach(function(key) {
+    node = node[key];
+  });
+
```

Comment:
And please notice that due to the check in loopback-datasource-juggler: https://github.com/strongloop/loopback-datasource-juggler/blob/839bb9a81eac236f7356651881def0c67088e007/lib/dao.js#L1793-L1796
function `buildNearFilter` will NOT be called when it's with logic operators. 
Because `geo.nearFilter(query.where);` doesn't check operators.